src/Native/System.Native/pal_networking.cpp

```diff
@@ -2013,3 +2018,21 @@ extern "C" Error WaitForSocketEvents(int32_t port, SocketEvent* buffer, int32_t*
 
     return WaitForSocketEventsInner(port, buffer, count);
 }
+
+extern "C" int32_t PlatformSupportsMultipleConnectAttempts()
+{
+#if defined(PLATFORM_SUPPORTS_MULTIPLE_CONNECT_ATTEMPTS)
```

Comment:
@jasonwilliams200OK: my best guess is that this feature is not supported on FreeBSD. POSIX states that the state of a socket after a failed connect is unspecified, so the only truly POSIX-compliant approach to multiple connect attempts is to close the current socket after a failed connect and open a new socket for the next attempt[1].

[1] http://pubs.opengroup.org/onlinepubs/9699919799/functions/connect.html